Friend/firmware/firmware_v1.0/src/transport.c

```diff
@@ -65,7 +68,119 @@ static struct bt_gatt_attr dfu_service_attr[] = {
 };
 
 static struct bt_gatt_service dfu_service = BT_GATT_SERVICE(dfu_service_attr);
+//Acceleration data
+//this code activates the onboard accelerometer. some cute ideas may include shaking the necklace to color strobe
+//
+typedef struct sensors{
+
+	struct sensor_value a_x;
+	struct sensor_value a_y;
+	struct sensor_value a_z;
+    struct sensor_value g_x;
+    struct sensor_value g_y;
+    struct sensor_value g_z;
+
+};
+static struct sensors mega_sensor;
+static struct device *lsm6dsl_dev;
+//Arbritrary uuid, feel free to change
+static struct bt_uuid_128 accel_uuid = BT_UUID_INIT_128(BT_UUID_128_ENCODE(0x00000000,0x0000,0x1000,0x7450,0xBE2E44B06B00));
```

Comment:
How did you generate this UUID? Looks... suspicious to me :-)
Also, any particular reason we want this in a separate service?

Reply:
I was following the conventions that appeared to be in the file already. The contributors separated the services such as battery, firmware ota, and audio, so I thought it would be appropriate to make a new service too. I suppose you could stuff all data into a single service labeled "friends", although I'm unsure if it is possible.

Reply:
We have separate services, because Battery Service and Device Information Service are well-known services that are defined on the BLE standard with 16-bit service IDs.

The OTA is a custom service as well, but it is a well-known Nordic service, so it is a de-facto industry standard.

The audio and video data streaming are custom logic specific to our device. So are the accelerometer and button. We could be putting these into separate services, but honestly the only reason would be to make it easy to build devices that have accelerometer/button, but don't have audio/video streaming, which I don't see happening.

Separately, there's a limitation on the number of custom services that can be advertised in the broadcast advertising packets, which makes the discovery of multiple services bit more complex than a single service. This has to do with the fact that the space in the advertising packets is limited, and custom services have 128-bit UUIDs.

And from app developers POV, it might be easier to maintain state transition if dealing with a single service with multiple characteristics, than with multiple services.

Honestly, neither of these is a strong argument against separate service on its own, but all together make me lean towards merging accelerometer/button into the main service together with the audio/video data.

But I am not going to fight this too hard, as long as the actual logic works :-)